.github/workflows/main.yml

```diff
@@ -1,7 +1,15 @@
 name: CI
 
 # Controls when the action will run.
-on: [push, pull_request, workflow_dispatch]
+on:
+  # Triggers the workflow on push or pull request events but only for the main branch
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
```

Comment:
What about adding something like `if: github.repository != ‘adsharma/py2many’ || github.event.base.ref == 'main'` instead to the jobs?

Reply:
That seems workable. Do you want to submit a new PR? I'll test it this time :)